public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -142,3 +144,34 @@ export function alertStateToReadable(state: PromAlertingRuleState | GrafanaAlert
   }
   return capitalize(state);
 }
+
+export type FlatRule = {
+  dataSourceName: string;
+  namespaceName: string;
+  groupName: string;
+} & AlertingRule;
+
+export const flattenRules = (rules: RuleNamespace[]) => {
+  return rules.reduce((acc, { dataSourceName, name: namespaceName, groups }) => {
+    groups.forEach(({ name: groupName, rules }) => {
+      rules.forEach((rule) => {
+        if (rule.type !== 'recording' && (rule.state === 'firing' || rule.state === 'pending')) {
```

Comment:
```suggestion
        if (isAlertingRule(rule) && (rule.state === PromAlertingRuleState.Firing || rule.state === PromAlertingRuleState.Pending)) {
```

Reply:
I've now added filters options for each of these.